### PR TITLE
Update release and environment details

### PR DESCRIPTION
**Pull request by collinpreston**
### Description (What does it do?)
Updates the instructions for the release process.  Updates the details regarding the environments.

### Additional Context
There is a lot in the handbook that is out of date and no longer applicable.  I don't know how often the handbook is being used by other developers. 

We should gauge how much the handbook is being used.  If it still plays a role within our team, we should remove much of the information and start fresh.

### CODE REVIEW

File: delivering/release-process/index.md [hunk 1]
```diff
@@ -5,64 +5,42 @@ has_children: true
...
 
-- The main channel for communication around releases is the Slack channel for the product in question.
-- Engineers are free to augment Slack with other tools (e.g. Zoom) as they see fit.
-- If you won't be available to verify your changes in RC, communicate this to the team and ask a team member to do this in your absence before merging changes.
-- If a release candidate isn't viable for release, communicate this and what the next step(s) are.
-
-### Individual Engineer
+- The main channel for communication around releases is the Slack channel for the product associated with the release.
+- If you won't be available to verify your changes in the RC environment, ask a team member to take over and verify your changes in RC.
+- If a release candidate should not be promoted to the Production environment, identify and communicate the next steps within the Slack channel for the product associated with the release.s
```
Comment (pdpinch) [inline delivering/release-process/index.md:16]: typo
```suggestion
- If a release candidate should not be promoted to the Production environment, identify and communicate the next steps within the Slack channel for the product associated with the release.
```

File: delivering/release-process/webapp-release-process.md [hunk 2]
```diff
@@ -9,28 +9,28 @@ title: Web Applications
...
 
 
 - Before your code is merged:
   - Coordinate with DevOps on having configuration in place both on RC and production for your changes _before_ it gets merged so that releases are not held up.
 - After your code is merged:
-  - When the changes have been deployed on RC, verify your changes with a round of functional testing.
+  - When the changes have been deployed on RC, verify your changes via functional testing and obtain acceptance from any necessary stakeholders.
 - After your code is released:
-  - Perform smoke testing of the functionality if it is prudent and won't disrupt other users.
+  - Perform smoke testing of the functionality if it won't interfere other application users.
```
Comment (pdpinch) [inline delivering/release-process/webapp-release-process.md:20]:
```suggestion
  - Perform smoke testing of the functionality if it won't interfere with other application users.
```

File: delivering/release-process/webapp-release-process.md [hunk 3]
```diff
@@ -9,28 +9,28 @@ title: Web Applications
...
 
 ### Engineering Team
 
-Collectively the team members who have made changes are responsible for coordinating the release to production:
+Collectively, the team members who have made changes are responsible for coordinating the release to production:
 
-- The team members who have changes in the release candidate are responsible for verifying their changes:
+- The team members who have changes to the main branch are responsible for verifying their changes:
```
Comment (pdpinch) [inline delivering/release-process/webapp-release-process.md:26]: Why did you make this change? It seems to me only those with commits in the RC can verify their changes.

**Review by pdpinch — APPROVED** [review]
Thanks for making this update. Based on the comments on https://github.com/mitodl/hq/discussions/404 I thought these changes had already been made to the handbook. 😳

File: delivering/release-process/webapp-release-process.md (hunk 3 above)
Comment (collinpreston) [reply delivering/release-process/webapp-release-process.md:26]: the documentation does not provided a clear explanation that "release candidate" is referring to a branch within Github and not the release itself.

**Pull request merged**